src/main/java/refactoringml/db/Database.java

```diff
@@ -1,144 +1,106 @@
 package refactoringml.db;
```

Comment:
Can you please transfer the changes to the Database connection and exception handling into a new PR, because it is unrelated to the other quality of life changes. Also, I am not certain how some of the changes behave in specific situations, e.g. store CommitMetaData in db -> store related RefactoringInstance in db -> Exception -> delete CommitMetaData from db -> Foreign Key Constraint Error on db?